ballerina-shell/modules/shell-core/src/main/java/io/ballerina/shell/parser/SerialTreeParser.java

```diff
@@ -75,6 +76,10 @@ public Node parse(String source) throws TreeParserException {
             } catch (ParserRejectedException e) {
                 errorMessage = "Invalid statement: " + e.getMessage();
                 break;
+            } catch (InvalidMethodException e) {
+                errorMessage = e.getMessage();
+                addErrorDiagnostic(errorMessage);
+                throw new TreeParserException();
```

Comment:
What's the reason for throwing another exception here? 🤔

Reply:
Removed in 5732df29120212bc9c5639fb1901b98d6172d290

Reply:
No I meant, was that added with a reason? If it was added for a reason, it's ok. Otherwise no point since the same exception is thrown at the end of the method.